WireDomain/Sources/WireDomain/Repositories/User/UserRepository.swift

```diff
@@ -321,27 +339,113 @@ public final class UserRepository: UserRepositoryProtocol {
         }
     }
 
+    public func fetchAllUserIdsWithOneOnOneConversation() async throws -> [WireDataModel.QualifiedID] {
```

Comment:
This return type might be problematic since we (should) only store domains if federation is enabled. Perhaps we could fallback to the local domain in case there is no domain stored in the db.